sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/MergingSessionsIterator.scala

```diff
@@ -118,7 +118,9 @@ class MergingSessionsIterator(
       val inputRow = inputIterator.next()
       nextGroupingKey = groupingWithoutSessionProjection(inputRow).copy()
       val session = sessionProjection(inputRow)
-      nextGroupingSession = session.getStruct(0, 2).copy()
+      val groupingSession = session.getStruct(0, 2)
```

Comment:
1. We have another place to have the same possibility of NPE, processCurrentSortedGroup(). We may want to extract the logic to apply the same to both places.
2. I guess it's OK to set errorOnIterator = true and throw internalError "here" instead of wrapping the call of initialize() with try-catch.